currency/storage_test.go

```diff
@@ -145,4 +145,67 @@ func TestRunUpdater(t *testing.T) {
 	if !errors.Is(err, nil) {
 		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
 	}
+
+	err = newStorage.Shutdown()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// old config where two providers enabled
+	other := settings
+	settings.Name = "ExchangeRates"
+	settings.Enabled = true
+	settings.APIKey = "" // old default provider which did not need api keys.
+	settings.PrimaryProvider = true
+	other.Name = "OpenExchangeRates" // Has keys enabled and will fall over to primary
```

Comment:
This comment is not accurate:
![image](https://user-images.githubusercontent.com/9261323/154400721-ccae98d6-df02-4372-8aba-a2691afd8376.png)

You also do not check if it is primary